test/e2e/kubectl.go

```diff
@@ -242,42 +172,9 @@ func getData(c *client.Client, podID string) (*updateDemoData, error) {
 	Logf("got data: %s", body)
 	var data updateDemoData
 	err = json.Unmarshal(body, &data)
```

Comment:
Check the error after unmarshaling, e.g. if err := json.Unmarshal(...) err != nil { return nil, err }